lib/BuildSystem/BuildSystem.cpp

```diff
@@ -1203,6 +1205,13 @@ BuildSystemImpl::lookupNode(StringRef name, bool isImplicit) {
 }
 
 llvm::Optional<BuildValue> BuildSystemImpl::build(BuildKey key) {
+
+  // If we were cancelled, return.
+  if (isCancelled_) {
+      isCancelled_ = false;
```

Comment:
Maybe BuildSystem should have its own reset method and frontend delegate should call that if it has a build system?